src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -2786,225 +2719,333 @@ namespace HeatBalanceSurfaceManager {
-                } else {
-                    InShelfSurf = 0;
-                    OutShelfSurf = 0;
-                }
-
-                if (Surface(SurfNum).ExtSolar || SurfWinOriginalClass(SurfNum) == SurfaceClass_TDD_Diffuser) {
-
-                    if (SurfWinOriginalClass(SurfNum) == SurfaceClass_TDD_Diffuser) {
-                        PipeNum = SurfWinTDDPipeNum(SurfNum);
-                        SurfNum2 = TDDPipe(PipeNum).Dome;
-
-                        CosInc = CosIncAng(TimeStep, HourOfDay, SurfNum2);
-
-                        // Reconstruct the beam, sky, and ground radiation transmittance of just the TDD:DOME and TDD pipe
-                        // by dividing out diffuse solar transmittance of TDD:DIFFUSER
-                        BeamSolar = BeamSolarRad * TransTDD(PipeNum, CosInc, SolarBeam) / dataConstruction.Construct(ConstrNum).TransDiff;
-
-                        SkySolarInc = DifSolarRad * AnisoSkyMult(SurfNum2) * TransTDD(PipeNum, CosInc, SolarAniso) /
-                                      dataConstruction.Construct(ConstrNum).TransDiff;
-
-                        GndSolarInc = GndSolarRad * Surface(SurfNum2).ViewFactorGround * TDDPipe(PipeNum).TransSolIso /
-                                      dataConstruction.Construct(ConstrNum).TransDiff;
-
-                    } else if (OutShelfSurf > 0) { // Outside daylighting shelf
-                        SurfNum2 = SurfNum;
-
-                        CosInc = CosIncAng(TimeStep, HourOfDay, SurfNum);
-
-                        BeamSolar = BeamSolarRad;
-                        SkySolarInc = DifSolarRad * AnisoSkyMult(SurfNum);
-
-                        ShelfSolarRad = (BeamSolarRad * SunlitFrac(TimeStep, HourOfDay, OutShelfSurf) * CosIncAng(TimeStep, HourOfDay, OutShelfSurf) +
-                                         DifSolarRad * AnisoSkyMult(OutShelfSurf)) *
-                                        Shelf(ShelfNum).OutReflectSol;
-
-                        // Add all reflected solar from the outside shelf to the ground solar
-                        // NOTE:  If the shelf blocks part of the view to the ground, the user must reduce the ground view factor!!
-                        GndSolarInc = GndSolarRad * Surface(SurfNum).ViewFactorGround + ShelfSolarRad * Shelf(ShelfNum).ViewFactor;
-
-                    } else { // Regular surface
-                        SurfNum2 = SurfNum;
-                        CosInc = CosIncAng(TimeStep, HourOfDay, SurfNum);
-                        BeamSolar = BeamSolarRad;
-                        SkySolarInc = SurfSkySolarInc(SurfNum);
-                        GndSolarInc = SurfGndSolarInc(SurfNum);
-                    }
-
+            if (BuildingShadingCount || FixedShadingCount || AttachedShadingCount) {
+                for (int SurfNum = 1; SurfNum <= TotSurfaces; ++SurfNum) {
+                    if (!Surface(SurfNum).ShadowingSurf) continue;
```

Comment:
I'm actually working on this at the other branch, but haven't fully tested this piece yet. Do we want to pull that piece in this one? I was hoping this can go first so I'm not blocking Matt.

Reply:
This is ok here. It can wait for your next phase.